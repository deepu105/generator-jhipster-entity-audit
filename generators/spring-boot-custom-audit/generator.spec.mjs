import { beforeAll, describe, expect, it } from 'vitest';

import { defaultHelpers as helpers, result } from 'generator-jhipster/testing';

const SUB_GENERATOR = 'spring-boot-custom-audit';
const SUB_GENERATOR_NAMESPACE = `jhipster-entity-audit:${SUB_GENERATOR}`;

describe('SubGenerator spring-boot-custom-audit of entity-audit JHipster blueprint', () => {
  describe('run', () => {
    beforeAll(async function () {
      await helpers
        .run(SUB_GENERATOR_NAMESPACE)
        .withJHipsterConfig()
        .withOptions({
          creationTimestamp: '2022-01-01',
          ignoreNeedlesError: true,
        })
        .withJHipsterLookup()
        .withParentBlueprintLookup();
    });

    it('should succeed', () => {
      expect(result.getStateSnapshot()).toMatchSnapshot();
    });
  });
});
