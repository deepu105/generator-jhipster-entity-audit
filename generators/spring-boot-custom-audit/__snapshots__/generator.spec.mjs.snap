// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SubGenerator spring-boot-custom-audit of entity-audit JHipster blueprint > run > should succeed 1`] = `
{
  ".jhipster/Audited.json": {
    "stateCleared": "modified",
  },
  ".yo-rc.json": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/GeneratedByJHipster.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/audit/AsyncEntityAuditEventWriter.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/audit/EntityAuditEventListener.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/audit/EntityAuditEventWriter.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/audit/package-info.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/domain/Audited.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/domain/EntityAuditEvent.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/domain/enumeration/EntityAuditAction.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/domain/enumeration/package-info.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/domain/package-info.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/package-info.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/repository/EntityAuditEventRepository.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/repository/package-info.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/web/rest/EntityAuditResource.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/web/rest/package-info.java": {
    "stateCleared": "modified",
  },
  "src/main/resources/config/liquibase/changelog/20220101000100_added_entity_EntityAuditEvent.xml": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/audit/TestEntityAuditEventWriter.java": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/domain/AuditedTest.java": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/domain/AuditedTestSamples.java": {
    "stateCleared": "modified",
  },
}
`;
